x/gamm/keeper/msg_server_test.go

```diff
@@ -0,0 +1,226 @@
+package keeper_test
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/keeper"
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/types"
+)
+
+const (
+	doesNotExistDenom = "nodenom"
+)
+
+var _ = suite.TestingSuite(nil)
+
+// TestSwapExactAmountIn_Events tests that events are correctly emitted
+// when calling SwapExactAmountIn.
+func (suite *KeeperTestSuite) TestSwapExactAmountIn_Events() {
+	const (
+		tokenInMinAmount = 1
+		tokenIn          = 5
+	)
+
+	testcases := map[string]struct {
+		routes                []types.SwapAmountInRoute
+		tokenIn               sdk.Coin
+		tokenOutMinAmount     sdk.Int
+		expectError           bool
```

Comment:
This should probably be flipped to `expectPass` to be in line with the rest of the table driven tests in our codebase. If there are any other `expectError`s, we should probably flip them to expectPass (or vice versa) for consistency (I'll make a separate issue about making this switch for other files)

Reply:
The best practice and a convention is to have `expectError error` where we can assert the actual kind of the error. Using a boolean is a shortcut to the original rule.

I'm not sure why we would have `expectPass` as a rule. I think it might make sense where the majority of test cases are errors and only one is valid. However, I think this can be decided on the case-by-case basis